examples/layer-browser/src/examples/additional-layers.js

```diff
@@ -63,26 +65,18 @@ const SimpleMeshLayerExample = {
 
 const ScenegraphLayerExample = {
   layer: ScenegraphLayer,
-  initialize: () => {
-    const url =
-      'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/Duck/glTF-Binary/Duck.glb';
-    window
-      .fetch(url)
-      .then(res => res.arrayBuffer())
-      .then(data => {
-        const gltfParser = new GLTFParser();
-        ScenegraphLayerExample.props.gltf = gltfParser.parse(data);
-      });
-  },
   props: {
     id: 'scenegraph-layer',
     data: dataSamples.points,
     pickable: true,
     sizeScale: 1,
+    scenegraph:
+      'https://raw.githubusercontent.com/KhronosGroup/glTF-Sample-Models/master/2.0/Duck/glTF-Binary/Duck.glb',
+    fetch: async (url, layer) => (await loadFile(url, layer.getLoadOptions())).scenes[0],
```

Comment:
`data` is also async, and this will break using an URL for `data`.

If `@loaders.gl/core` is already a dependency, we can offer a better `fetch` as the default prop of `ScenegraphLayer`, something like:

```js
fetch: (url, layer) => {
  if (url.endsWidth('json')) {
	return fetch(url).then(resp => resp.json());
  }
  return loadFile(url)...
}
```

Reply:
`@loaders.gl/core` is a dependency for mesh-layers but not deck.gl/core.

Reply:
I have a better solution: pass the prop name as fetch parameter.

Reply:
> @loaders.gl/core is a dependency for mesh-layers but not deck.gl/core

You can override the default `fetch` of the `Layer` class in `ScenegraphLayer.defaultProps`.

Reply:
With the current logic mesh layers do not depend on `@loaders.gl/core`.